src/lib/device/chunk_selector.hpp

```diff
@@ -0,0 +1,58 @@
+/*********************************************************************************
+ * Modifications Copyright 2017-2019 eBay Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <vector>
+#include <folly/ThreadLocal.h>
+#include <sisl/logging/logging.h>
+
+#include <homestore/homestore_decl.hpp>
+#include <homestore/blk.h>
+
+namespace homestore {
+class Chunk;
+
+class ChunkSelector {
```

Comment:
Absolutely, we can add it as part of heapchunkselector PR.

Reply:
> should we add a `remove` method to remove a chunk from a selector. for example, if a HDD is broken , we may want to remove all the chunk residing in it

Yes absolutely, but it can be done as part of HeapChunkSelector PR

Reply:
since the virtual class of chunkselector is defined in homestore here, please add this method to the virtual class in this PR ,and HeapChunkSelector will implement it in homeObject.

Reply:
We can add any remaining interfaces as part of HeapChunkSelector PR, but that should only be limited to chunk operation (add/remove) etc inside the base class ChunkSelector.